File: LeetCode_ProblemSet/Go/Problem3016.go
package main

import "sort"

func minimumPushes(word string) int {
	count := map[rune]int{}
	for _, ch := range word {
		count[ch]++
	}

	keys := []rune{}
	for ch := range count {
		keys = append(keys, ch)
	}
	sort.SliceStable(keys, func(i, j int) bool {
		return count[keys[j]] < count[keys[i]]
	})

	cost := 0
	for idx, ch := range keys {
		if idx < 8 {
			cost += count[ch]
		} else if idx < 16 {
			cost += count[ch] * 2
		} else if idx < 24 {
			cost += count[ch] * 3
		} else {
			cost += count[ch] * 4
		}
	}
	return cost
}

File: LeetCode_ProblemSet/Problem3016.py
class Solution(object):
    def minimumPushes(self, word):
        """
        :type word: str
        :rtype: int
        """
        count = {}
        for ch in word:
            count[ch] = count.get(ch, 0) + 1
            
        keys = [key for key in count]
        keys = sorted(keys, key=lambda ch: count[ch], reverse=True)
        
        cost = 0
        for idx in range(len(keys)):
            if idx < 8:
                cost += count[keys[idx]]
            elif idx < 16:
                cost += count[keys[idx]] * 2
            elif idx < 24:
                cost += count[keys[idx]] * 3
            else:
                cost += count[keys[idx]] * 4
                
        return cost